ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -424,10 +424,15 @@ func (r *RayServiceReconciler) reconcileRayCluster(ctx context.Context, rayServi
 	if clusterAction == RolloutNew {
 		// For LLM serving, some users might not have sufficient GPU resources to run two RayClusters simultaneously.
 		// Therefore, KubeRay offers ENABLE_ZERO_DOWNTIME as a feature flag for zero-downtime upgrades.
+		zeroDowntimeEnvVar := os.Getenv(ENABLE_ZERO_DOWNTIME)
+		rayServiceSpecUpgradeStrategy := rayServiceInstance.Spec.UpgradeStrategy
 		enableZeroDowntime := true
-		if s := os.Getenv(ENABLE_ZERO_DOWNTIME); strings.ToLower(s) == "false" {
-			enableZeroDowntime = false
+		if rayServiceSpecUpgradeStrategy != "" {
+			enableZeroDowntime = rayServiceSpecUpgradeStrategy == rayv1.BlueGreenUpgrade
+		} else if zeroDowntimeEnvVar != "" {
+			enableZeroDowntime = strings.ToLower(zeroDowntimeEnvVar) == "true"
```

Comment:
`strings.ToLower(zeroDowntimeEnvVar) == "true"` and `strings.ToLower(zeroDowntimeEnvVar) != "false"` are not equivalent.

```suggestion
			enableZeroDowntime = strings.ToLower(zeroDowntimeEnvVar) != "false"
```

Reply:
Ahh thats my confusion, I had a different notion based on explicit "true"